<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="An accurate and easy tool for performing linear trajectory inference on
  single cells using single-cell RNA sequencing data. In addition, SCORPIUS
  provides functions for discovering the most important genes with respect to
  the reconstructed trajectory, as well as nice visualisation tools.
  Cannoodt et al. (2016) &lt;doi:10.1101/079509&gt;.">
<title>Inferring Developmental Chronologies from Single-Cell RNA Sequencing Data • SCORPIUS</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Inferring Developmental Chronologies from Single-Cell RNA Sequencing Data">
<meta property="og:description" content="An accurate and easy tool for performing linear trajectory inference on
  single cells using single-cell RNA sequencing data. In addition, SCORPIUS
  provides functions for discovering the most important genes with respect to
  the reconstructed trajectory, as well as nice visualisation tools.
  Cannoodt et al. (2016) &lt;doi:10.1101/079509&gt;.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">SCORPIUS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.8</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/anndata.html">Running SCOPIUS on an AnnData object</a>
    <a class="dropdown-item" href="articles/ginhoux.html">Investigating dendritic cell maturation in dendritic cell progenitors</a>
    <a class="dropdown-item" href="articles/seurat.html">Running SCOPIUS on a Seurat object</a>
    <a class="dropdown-item" href="articles/simulated-data.html">Trajectory inference from simulated data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/rcannood/SCORPIUS/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="scorpius">SCORPIUS<a class="anchor" aria-label="anchor" href="#scorpius"></a>
</h1></div>
<!-- badges: start -->

<p>SCORPIUS an unsupervised approach for inferring linear developmental chronologies from single-cell RNA sequencing data. In comparison to similar approaches, it has three main advantages:</p>
<ul>
<li><p><strong>It accurately reconstructs linear dynamic processes.</strong> The performance was evaluated using a quantitative evaluation pipeline and ten single-cell RNA sequencing datasets.</p></li>
<li><p><strong>It automatically identifies marker genes, speeding up knowledge discovery.</strong></p></li>
<li><p><strong>It is fully unsupervised.</strong> Prior knowledge of the relevant marker genes or cellular states of individual cells is not required.</p></li>
</ul>
<p>News:</p>
<ul>
<li><p>See <code>news(package = "SCORPIUS")</code> for a full list of changes to the package.</p></li>
<li><p>Our preprint is on <a href="https://biorxiv.org/content/early/2016/10/07/079509" class="external-link">bioRxiv</a> (Cannoodt et al. 2016).</p></li>
<li><p>Check out our <a href="https://dx.doi.org/10.1038/s41587-019-0071-9" class="external-link">review</a> on Trajectory Inference methods (Saelens et al. 2019).</p></li>
</ul>
<div class="section level2">
<h2 id="installing-scorpius">Installing SCORPIUS<a class="anchor" aria-label="anchor" href="#installing-scorpius"></a>
</h2>
<p>You can install:</p>
<ul>
<li>
<p>the latest released version from CRAN with</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"SCORPIUS"</span><span class="op">)</span></span></code></pre></div>
</li>
<li>
<p>the latest development version from GitHub with</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"rcannood/SCORPIUS"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</li>
</ul>
<p>If you encounter a bug, please file a minimal reproducible example on the <a href="https://github.com/rcannood/SCORPIUS/issues" class="external-link">issues</a> page.</p>
</div>
<div class="section level2">
<h2 id="learning-scorpius">Learning SCORPIUS<a class="anchor" aria-label="anchor" href="#learning-scorpius"></a>
</h2>
<p>To get started, read the introductory example below, or read one of the vignettes containing more elaborate examples:</p>
<ul>
<li>Running SCOPIUS on an AnnData object:<br><code><a href="articles/anndata.html">vignette("anndata", package="SCORPIUS")</a></code>
</li>
<li>Investigating dendritic cell maturation in dendritic cell progenitors:<br><code><a href="articles/ginhoux.html">vignette("ginhoux", package="SCORPIUS")</a></code>
</li>
<li>Running SCOPIUS on a Seurat object:<br><code><a href="articles/seurat.html">vignette("seurat", package="SCORPIUS")</a></code>
</li>
<li>Trajectory inference from simulated data:<br><code><a href="articles/simulated-data.html">vignette("simulated-data", package="SCORPIUS")</a></code>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="introductory-example">Introductory example<a class="anchor" aria-label="anchor" href="#introductory-example"></a>
</h2>
<p>This section describes the main workflow of SCORPIUS without going in depth in the R code. For a more detailed explanation, see the vignettes listed below.</p>
<p>To start using SCORPIUS, simply write:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rcannood/SCORPIUS" class="external-link">SCORPIUS</a></span><span class="op">)</span></span></code></pre></div>
<p>The <code>ginhoux</code> dataset (Schlitzer et al. 2015) contains 248 dendritic cell progenitors in one of three cellular cellular states: MDP, CDP or PreDC. Note that this is a reduced version of the dataset, for packaging reasons. See ?ginhoux for more info.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ginhoux</span><span class="op">)</span></span>
<span><span class="va">expression</span> <span class="op">&lt;-</span> <span class="va">ginhoux</span><span class="op">$</span><span class="va">expression</span></span>
<span><span class="va">group_name</span> <span class="op">&lt;-</span> <span class="va">ginhoux</span><span class="op">$</span><span class="va">sample_info</span><span class="op">$</span><span class="va">group_name</span></span></code></pre></div>
<p>With the following code, SCORPIUS reduces the dimensionality of the dataset and provides a visual overview of the dataset. In this plot, cells that are similar in terms of expression values will be placed closer together than cells with dissimilar expression values.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">space</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/reduce_dimensionality.html">reduce_dimensionality</a></span><span class="op">(</span><span class="va">expression</span>, <span class="st">"spearman"</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/draw_trajectory_plot.html">draw_trajectory_plot</a></span><span class="op">(</span><span class="va">space</span>, <span class="va">group_name</span>, contour <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README_reduce_dimensionality-1.png"><!-- --></p>
<p>To infer and visualise a trajectory through these cells, run:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">traj</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/infer_trajectory.html">infer_trajectory</a></span><span class="op">(</span><span class="va">space</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/draw_trajectory_plot.html">draw_trajectory_plot</a></span><span class="op">(</span><span class="va">space</span>, <span class="va">group_name</span>, <span class="va">traj</span><span class="op">$</span><span class="va">path</span>, contour <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README_infer_trajectory-1.png"><!-- --></p>
<p>To identify candidate marker genes, run:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># warning: setting num_permutations to 10 requires a long time (~30min) to run!</span></span>
<span><span class="co"># set it to 0 and define a manual cutoff for the genes (e.g. top 200) for a much shorter execution time.</span></span>
<span><span class="va">gimp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/gene_importances.html">gene_importances</a></span><span class="op">(</span></span>
<span>  <span class="va">expression</span>, </span>
<span>  <span class="va">traj</span><span class="op">$</span><span class="va">time</span>, </span>
<span>  num_permutations <span class="op">=</span> <span class="fl">10</span>, </span>
<span>  num_threads <span class="op">=</span> <span class="fl">8</span>, </span>
<span>  ntree <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  ntree_perm <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="op">)</span> </span></code></pre></div>
<p>To select the most important genes and scale its expression, run:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gimp</span><span class="op">$</span><span class="va">qvalue</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">gimp</span><span class="op">$</span><span class="va">pvalue</span>, <span class="st">"BH"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gimp</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gene_sel</span> <span class="op">&lt;-</span> <span class="va">gimp</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="va">gimp</span><span class="op">$</span><span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">.05</span><span class="op">]</span></span>
<span><span class="va">expr_sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dynutils/man/scale_quantile.html" class="external-link">scale_quantile</a></span><span class="op">(</span><span class="va">expression</span><span class="op">[</span>,<span class="va">gene_sel</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>To visualise the expression of the selected genes, use the <code>draw_trajectory_heatmap</code> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/draw_trajectory_heatmap.html">draw_trajectory_heatmap</a></span><span class="op">(</span><span class="va">expr_sel</span>, <span class="va">traj</span><span class="op">$</span><span class="va">time</span>, <span class="va">group_name</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README_visualise_tafs-1.png"><!-- --></p>
<p>Finally, these genes can also be grouped into modules as follows:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/extract_modules.html">extract_modules</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/dynutils/man/scale_quantile.html" class="external-link">scale_quantile</a></span><span class="op">(</span><span class="va">expr_sel</span><span class="op">)</span>, <span class="va">traj</span><span class="op">$</span><span class="va">time</span>, verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/draw_trajectory_heatmap.html">draw_trajectory_heatmap</a></span><span class="op">(</span><span class="va">expr_sel</span>, <span class="va">traj</span><span class="op">$</span><span class="va">time</span>, <span class="va">group_name</span>, <span class="va">modules</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README_moduled_tafs-1.png"><!-- --></p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Cannoodt2016" class="csl-entry">
Cannoodt, Robrecht, Wouter Saelens, Dorine Sichien, Simon Tavernier, Sophie Janssens, Martin Guilliams, Bart Lambrecht, Katleen De Preter, and Yvan Saeys. 2016. “SCORPIUS Improves Trajectory Inference and Identifies Novel Modules in Dendritic Cell Development,” October. <a href="https://doi.org/10.1101/079509" class="external-link uri">https://doi.org/10.1101/079509</a>.
</div>
<div id="ref-Saelens2019" class="csl-entry">
Saelens, Wouter, Robrecht Cannoodt, Helena Todorov, and Yvan Saeys. 2019. “A Comparison of Single-Cell Trajectory Inference Methods.” <em>Nature Biotechnology</em> 37 (5): 547–54. <a href="https://doi.org/10.1038/s41587-019-0071-9" class="external-link uri">https://doi.org/10.1038/s41587-019-0071-9</a>.
</div>
<div id="ref-Schlitzer2015" class="csl-entry">
Schlitzer, Andreas, V Sivakamasundari, Jinmiao Chen, Hermi Rizal Bin Sumatoh, Jaring Schreuder, Josephine Lum, Benoit Malleret, et al. 2015. “Identification of <span class="nocase">cDC</span>1- and <span class="nocase">cDC</span>2-Committed DC Progenitors Reveals Early Lineage Priming at the Common DC Progenitor Stage in the Bone Marrow.” <em>Nature Immunology</em> 16 (7): 718–28. <a href="https://doi.org/10.1038/ni.3200" class="external-link uri">https://doi.org/10.1038/ni.3200</a>.
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=SCORPIUS" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/rcannood/SCORPIUS/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/rcannood/SCORPIUS/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing SCORPIUS</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Robrecht Cannoodt <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-3641-729X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/rcannood/SCORPIUS/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/rcannood/SCORPIUS/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://cran.r-project.org/package=SCORPIUS" class="external-link"><img src="https://www.r-pkg.org/badges/version/SCORPIUS" alt="CRAN_Status_Badge"></a></li>
<li><a href="https://app.codecov.io/gh/rcannood/SCORPIUS?branch=master" class="external-link"><img src="https://codecov.io/gh/rcannood/SCORPIUS/branch/master/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Robrecht Cannoodt.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
